mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/PostgresAttachmentModule.java

```diff
@@ -35,20 +35,22 @@ public interface PostgresAttachmentModule {
     interface PostgresAttachmentTable {
 
         Table<Record> TABLE_NAME = DSL.table("attachment");
-        Field<UUID> ID = DSL.field("id", SQLDataType.UUID.notNull());
+        Field<UUID> ID_AS_UUID = DSL.field("id_as_uuid", SQLDataType.UUID.notNull());
+        Field<String> ID = DSL.field("id", SQLDataType.VARCHAR.notNull());
```

Comment:
So we need the "redundant" ID field just to pass IT? Any other way?

Reply:
If the tests fails because the structure is different, I would prefer here that we duplicate the test suite and adapt it to postgres attachment table structure instead?

Reply:
No, this is not "just to pass IT". In mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/mail/dao/PostgresAttachmentDAO.java line 87:

```
    public Flux<AttachmentMetadata> getAttachments(Collection<AttachmentId> attachmentIds) {
        return postgresExecutor.executeRows(dslContext -> Flux.from(dslContext.selectFrom(PostgresAttachmentTable.TABLE_NAME)
                .where(PostgresAttachmentTable.ID_AS_UUID.in(attachmentIds.stream().map(AttachmentId::asUUID).collect(ImmutableList.toImmutableList())))))
            .map(row -> AttachmentMetadata.builder()
                .attachmentId(AttachmentId.from(row.get(PostgresAttachmentTable.ID)))
                .type(row.get(PostgresAttachmentTable.TYPE))
                .messageId(PostgresMessageId.Factory.of(row.get(PostgresAttachmentTable.MESSAGE_ID)))
                .size(row.get(PostgresAttachmentTable.SIZE))
                .build());
    }
```

If we do not have ID_AS_UUID, we have to use a hashmap to save AttachmentId - UUID (UUID.nameUUIDFromBytes(id.getBytes(StandardCharsets.UTF_8))) which is not good option IMO

Reply:
Removed ID_AS_UUID. Use ID as pk (change type from UUID to VARCHAR)